packages/training/python/tests/e2e/test_online_training_e2e.py

```
@@ -0,0 +1,365 @@
+        assert bridge.base_url == "http://localhost:3001"
+        assert not bridge.is_initialized
+        assert bridge.npc_ids == []
+        assert bridge.archetypes == {}
+
+
+class TestOnlineEnvIntegration:
+    """Integration tests for online environment"""
+    
+    def test_pool_scenario_add_methods(self):
+        """Test Scenario.add_market, add_perpetual, add_news methods"""
+        from src.training.scenario_pool import Scenario, PortfolioState
```

Ruff (0.14.10)

Redefinition of unused `PortfolioState` from line 41

Remove definition: `PortfolioState`

(F811)